libbeat/logp/logger.go

```diff
@@ -1,31 +1,147 @@
 package logp
 
-import "fmt"
+import (
+	"go.uber.org/zap"
+)
 
-// Logger provides a logging type using the global logp functionality.
-// The Logger should be used to use with libraries havng a configurable logging
-// functionality.
+// LogOption configures a Logger.
+type LogOption = zap.Option
```

Comment:
I think this only works in Go 1.9, right? I think that's fine, just flagging in case this blocks some use case (like gccgo).

Reply:
Yes, type aliasing is only 1.9.

Should be OK in GCC8, not sure when is the release date.

__Due to the alignment of Go's semiannual release schedule with GCC's annual release schedule, GCC release 7 contains the Go 1.8.3 version of gccgo. We expect that the next release, GCC 8, will contain the Go 1.10 version of gccgo.__

Reply:
++, I think we're good to make use of it, as go 1.10 is in beta now so I imagine it's not that long until there's a GCC version that can work with it.